civcraft/src/com/avrgaming/civcraft/threading/sync/SyncGetChestInventory.java

```diff
@@ -57,26 +57,34 @@ public void run() {
 				for (int i = 0; i < UPDATE_LIMIT; i++) {
 					GetChestRequest request = requestQueue.poll();
 					if (request == null) {
-						return;
```

Comment:
This is really a big fix 👍

Reply:
https://github.com/ataranlen/civcraft/pull/170/commits/29050582d708004984459b6031a9d9d5b8d8a5fb